proto/umee/ugov/v1/query.proto

```diff
@@ -34,6 +36,41 @@ service Query {
   rpc InflationCycleEnd(QueryInflationCycleEnd) returns (QueryInflationCycleEndResponse) {
     option (google.api.http).get = "/umee/ugov/v1/inflation_cycle_end";
   }
+
+  // Token Balances queries for all account addresses that own a particular token
+  // denomination.
+  rpc TokenBalances(QueryTokenBalances) returns (QueryTokenBalancesResponse){
+    option (google.api.http).get = "/umee/ugov/v1/token_balances";
+  }
+}
+
+// which queries for a paginated set of all account holders of a particular
+// denomination.
+message QueryTokenBalances{
+  // denom defines the coin denomination to query all account holders for.
+  string denom = 1;
+  int64 height = 2;
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 3;
+}
+
+// QueryDenomOwnersResponse defines the RPC response of a TokenBalances RPC query.
+message QueryTokenBalancesResponse {
+  repeated TokenBalance token_balances = 1;
+   // pagination defines the pagination in the response.
+   cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// TokenBalance defines structure representing an account that owns or holds a
+// particular denominated token. It contains the account address and account
+// balance of the denominated token.
+//
+message TokenBalance {
+  // address defines the address that owns a particular denomination.
+  string address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+
+  // balance is the balance of the denominated coin for an account.
+  cosmos.base.v1beta1.Coin balance = 2 [(gogoproto.nullable) = false, (amino.dont_omitempty) = true];
```

Comment:
let's put `TokenBalance` into `QueryDenomOwnersResponse`

Reply:
I renamed token-balances to denom-owners